### DISCUSSION TITLE
Converting existing fields to translated fields..?

### DISCUSSION THREAD

**Discussion by MarkBurvs** [directus/directus#5573] [Archive]
We're using Directus as a website CMS, but now we're looking at localising all content for multiple languages.
We were hoping that we'd just be able to add translations to all existing fields, but unless we're missing something, it doesn't look like this is possible. Seems like the only way is to create new translation fields to replace all the existing fields, which, given the amount of content we have, is quite a big manual / error prone job.

Is there no way of converting all existing fields into translations fields, with all existing data automatically moved into a default language for each field? 

When I created a "translations" field in the Data Model editor, I was expecting to be able to maybe drag and drop existing fields onto the translations field to convert them, but doesn't seem this is possible..?

**Comment by benhaynes** [comment]
No, this isn't possible as of now. You would need to write a script using the API to migrate content from one field to the other.

**Comment by mfkrause** [comment]
If anyone stumbles across this in the future, here's a simple Python script that does this for you (just fill out the config fields on top). This requires that the old field names in the collection itself and the new field names in the translated relation are the same:

https://gist.github.com/mfkrause/50a106a4eac8d70928eb1ad27a2b64c8

**Comment by il-jakob** [comment]
Are there any plans to make create this feature? It would be very useful.

**Reply by rijkvanzanten** [reply]
Not as of right now.
